File: Phase5-TDD/InvertedIndex.Test/FileReaderTest.cs
using System;
using System.IO;
using System.Collections.Generic;
using Xunit;
using InvertedIndex.FileHandler;

namespace InvertedIndex.Test
{
    public class FileReaderTest : IDisposable
    {
        private const string BASE_PATH = "InvertedIndex.Test/docs";
        private const string TEST_FILE_NAME = "57110";
        private const string TEST_FILE_PATH1 = BASE_PATH + TEST_FILE_NAME;
        private readonly IList<string> TEST_FILE_WORDS = new List<string>() {"i","have","a","42","yr","old","male","friend"};
        private FileReader _fileReader;

        public FileReaderTest()
        {
            _fileReader = new FileReader();
        }

        [Fact]
        public void FileWordsReadingTest()
        {
            var file = new FileInfo(TEST_FILE_PATH1);
            var words = _fileReader.GetFileWords(file);

            Assert.Equal(TEST_FILE_WORDS, words);
        }

        [Fact]
        public void FileConvertToDocTest()
        {
            var docs = _fileReader.getFilesDocs(new FileInfo[] {new FileInfo(TEST_FILE_PATH1)});

            Assert.Equal(new List<Doc>()
            {
                new Doc(TEST_FILE_NAME, TEST_FILE_WORDS)
            }, docs);
        }

        public void Dispose()
        {
            GC.Collect();
        }
    }
}

File: Inverted-Index/src/tests/FileReaderTest.java
package tests;

import java.io.File;
import java.io.FileNotFoundException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;

import org.junit.After;
import org.junit.Assert;
import org.junit.Before;
import org.junit.Test;

import file_handler.Doc;
import file_handler.FileReader;

public class FileReaderTest {

    private final String BASE_PATH = "C:\\Repos\\Summer1400-SE-Team24\\Inverted-Index\\src\\tests\\docs\\";
    private final String TEST_FILE_NAME = "57110";
    private final String TEST_FILE_PATH1 = BASE_PATH + TEST_FILE_NAME;
    
    private final List<String> TEST_FILE_WORDS = Arrays.asList("i","have","a","42","yr","old","male","friend");
    private FileReader fileReader;

    @Before
    public void init() {
        fileReader = new FileReader();
    }

    @Test
    public void fileWordsReadingTest() throws FileNotFoundException {
        var file = new File(TEST_FILE_PATH1);
        var words = fileReader.getFilesWords(file);

        Assert.assertEquals(TEST_FILE_WORDS, words);
    }

    @Test
    public void fileConvertToDocTest() throws FileNotFoundException {
        var docs = fileReader.getFilesDocs(new File[] {new File(TEST_FILE_PATH1)});

        Assert.assertEquals(new ArrayList<>() {
            {add(new Doc(TEST_FILE_NAME, TEST_FILE_WORDS));}
        }, docs);
    }

    @After
    public void dispose() {
        fileReader = null;
    }
    
}
